CMakeLists.txt

```diff
@@ -659,15 +661,12 @@ if(UNIX)
     target_link_libraries(mxnet PUBLIC gcov)
   endif()
   if(CMAKE_BUILD_TYPE STREQUAL "Distribution")
-    # TODO For handling mxnet's symbols the following can be replace by
-    # annotating symbol visibility in source code, specifying
-    # set(CMAKE_CXX_VISIBILITY_PRESET hidden) and
-    # set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
-    # Symbols from statically linked libraries can be discarded via -Wl,--exclude-libs,ALL
-    if(APPLE)
-      set_target_properties(mxnet PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list,${PROJECT_SOURCE_DIR}/cmake/libmxnet.sym")
+    if(NOT APPLE)
+      # Do not export symbols from statically linked dependencies such as zmq, opencv, etc.
+      set_target_properties(mxnet PROPERTIES LINK_FLAGS "-Wl,--exclude-libs,ALL")
     else()
-      set_target_properties(mxnet PROPERTIES LINK_FLAGS "-Wl,--version-script=${PROJECT_SOURCE_DIR}/cmake/libmxnet.ver")
+      # Apple does not support -Wl,--exclude-libs,ALL
+      set_target_properties(mxnet PROPERTIES LINK_FLAGS "-Wl,-exported_symbols_list,${PROJECT_SOURCE_DIR}/cmake/libmxnet.sym")
```

Comment:
by not using a whitelist for non-apple, I think the resulting exported symbols would be different.

Reply:
Both options are equivalent given the naming convention in nnvm and mxnet